docs/articles/recommender_from_clickstreams.md

```diff
@@ -0,0 +1,214 @@
+# How to Build a Recommender System From Clickstream Data
+
+<!-- test comment -->
+
+Whenever you have a website with products that customers can browse, you can hardly save yourself from the flood of **clickstream data**. For example, a user of a hotel website might produce the following data when browsing different hotels:
+
+```mermaid
+graph LR;
+      id1[Tranquil Haven Resort]-->id2[Summit Vista Inn];
+      id2[Summit Vista Inn]-->id3[Azure Sands Hotel];
+      id3[Azure Sands Hotel]-->id4[Evergreen Retreat Lodge];
+```
+
+They start with the Tranquil Haven Resort, and end their browsing session with the Evergreen Retreat Lodge. We can do different things with this data now, one of them being finding **embeddings** for all hotels.
+
+:::hint
+Embeddings are numerical representations of categories that carry more meaning than a simple one-hot encoding. They are vectors of floats, and the length of this vector is called the embedding dimension.
+:::
+
+But what could these embeddings mean? Well, it depends on how the users interact with your website. In the special case of hotels, maybe you know this behavior from yourself or your partner: You want to stay close to some location and check out many hotels that look interesting to you.
+Now, if we assume that most customers behave like this, then the embeddings of similar hotels are similar. So, we could build a **recommender for similar hotels** with this!
```

Comment:
Specifically, what assumptions? 1 geographical proximity? 2 what else?
Do we need to make these assumptions... or can't we just say the fact that the user went from one selection to another connects those selections somehow (or, in the negative, didn't go from one to another), we don't need to know exactly why, but as long as their subsequent/preceding selections are not completely random, we can embed the fact of their subsequent/precedent selection, and use this as the basis of a recSys?

Reply:
If customers usually browse similar hotels, the embeddings learned will have the property that similar hotels have similar embeddings.

However, if customers browse very different hotels, maybe in the order they want to visit them, the embeddings will mean something different. Here, hotels having similar embeddings means that they are close within the customer's travel.

It's more about interpreting the embeddings. Of course, you can compute them all the time, but if you want to answer what the proximity of embedding means, you need to know more about the data.

Reply:
Just so I'm clear on your thinking, "If customers usually browse similar hotels"
1 what is "similar" about the hotels? same question for "if customers browse very different hotels"..
2 is this one customer or several customers browsing hotels?

Reply:
In the first case, I mean when customers decide to go to Paris for a weekend, and want to find a hotel there. They will the probably check out some hotels that might quite similar in a sense, maybe they are all romantic ones, or very cheap ones. Also, the location will be similar.

In the second case, I mean more like a single customer is doing a road trip planning. First, you search for a hotel in Paris for a night or two, then you want a hotel in a completely different place, maybe even with completely different characteristics (besides the location).

Reply:
Let's go back to your original text "Well, it depends on how the users interact with your website. In the special case of hotels, maybe you know this behavior from yourself or your partner: You want to stay close to some location and check out many hotels that look interesting to you.
Now, if we assume that most customers behave like this, then the embeddings of similar hotels are similar. "

What you have is clickstream data. Correct?
I'm trying to understand the process you are describing. Is the argument is that you start with clickstream data, train a model on it, creating embeddings. If that's true, then I'm trying to understand how these assumptions enter in. If assumptions enter in as additional data when training the model, then we need to know how. Currently the description is vague. 
If, on the other hand, your sole data source is clickstream data, which is what's described below, then it's not clear how these assumptions enter in, the role they play as a data source or in the model training.

Reply:
The assumptions don't matter for the training, it's rather about interpreting the relationship between the embeddings after the training.

We start with clickstream data, wherever it comes from. Then we train a model and end up with embeddings.

But what does it mean if the embeddings of 2 hotels are close? To answer this question, we have to know how the clickstreams are generated, i.e., how the users behave.

Reply:
And what's your source of data for how the users behave?

Reply:
You could either do an analysis (of how the hotels in a session relate to each other), or just ask a bunch of users, I would say.